Support for .NET 10.0.300 SDK

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Feature description

```text
updater | 2026/05/20 15:05:08 WARN   Error determining dependencies from `/home/dependabot/dependabot-updater/repo/src/Api/Api.csproj`:
STDOUT:
8.0.418 [/usr/local/dotnet/current/sdk]
9.0.311 [/usr/local/dotnet/current/sdk]
10.0.103 [/usr/local/dotnet/current/sdk]


STDERR:
The command could not be loaded, possibly because:
  * You intended to execute a .NET application:
      The application 'msbuild' does not exist or is not a managed .dll or .exe.
  * You intended to execute a .NET SDK command:
      A compatible .NET SDK was not found.

Requested SDK version: 10.0.300
global.json file: /home/dependabot/dependabot-updater/repo/global.json

Installed SDKs:

Install the [10.0.300] .NET SDK or update [/home/dependabot/dependabot-updater/repo/global.json] to match an installed SDK.
```

> [@Tsingis](https://github.com/Tsingis) can you provide a full log of the run? The updater pre-scans the repo for `global.json` files and installs the appropriate SDKs prior to the run so something must be going wrong there.

Repo's global.json

```json
{
  "sdk": {
    "version": "10.0.300",
    "rollForward": "latestMinor",
    "allowPrerelease": false
  },
  "test": {
    "runner": "Microsoft.Testing.Platform"
  }
}
```

```text
2026-05-20T22:07:30.9338033Z Current runner version: '2.334.0'
2026-05-20T22:07:30.9372821Z ##[group]Runner Image Provisioner
2026-05-20T22:07:30.9374215Z Hosted Compute Agent
2026-05-20T22:07:30.9375230Z Version: 20260213.493
2026-05-20T22:07:30.9376146Z Commit: 5c115507f6dd24b8de37d8bbe0bb4509d0cc0fa3
2026-05-20T22:07:30.9377260Z Build Date: 2026-02-13T00:28:41Z
2026-05-20T22:07:30.9378341Z Worker ID: {9264f295-b82f-493c-a0e6-03f6cc6c6536}
2026-05-20T22:07:30.9379399Z Azure Region: centralus
2026-05-20T22:07:30.9380242Z ##[endgroup]
2026-05-20T22:07:30.9382787Z ##[group]Operating System
2026-05-20T22:07:30.9383897Z Ubuntu
2026-05-20T22:07:30.9384777Z 24.04.4
2026-05-20T22:07:30.9385492Z LTS
2026-05-20T22:07:30.9386220Z ##[endgroup]
2026-05-20T22:07:30.9387250Z ##[group]Runner Image
2026-05-20T22:07:30.9388075Z Image: ubuntu-24.04
2026-05-20T22:07:30.9388871Z Version: 20260513.135.3
2026-05-20T22:07:30.9390959Z Included Software: https://github.com/actions/runner-images/blob/ubuntu24/20260513.135/images/ubuntu/Ubuntu2404-Readme.md
2026-05-20T22:07:30.9393703Z Image Release: https://github.com/actions/runner-images/releases/tag/ubuntu24%2F20260513.135
2026-05-20T22:07:30.9395315Z ##[endgroup]
2026-05-20T22:07:30.9397060Z ##[group]GITHUB_TOKEN Permissions
2026-05-20T22:07:30.9399968Z Contents: read
2026-05-20T22:07:30.9400794Z Metadata: read
2026-05-20T22:07:30.9401543Z Packages: read
2026-05-20T22:07:30.9402577Z ##[endgroup]
2026-05-20T22:07:30.9405349Z Secret source: None
2026-05-20T22:07:30.9406472Z Prepare workflow directory
2026-05-20T22:07:30.9761008Z Prepare all required actions
2026-05-20T22:07:30.9814521Z Getting action download info
2026-05-20T22:07:31.3474716Z Download action repository 'github/dependabot-action@main' (SHA:5fd0fae268a84cd4eba1c2af1d2628be708154c7)
2026-05-20T22:07:32.1766969Z Complete job name: Dependabot
2026-05-20T22:07:32.2616176Z ##[group]Run mkdir -p  ./dependabot-job-1377112841-1779314845
2026-05-20T22:07:32.2618024Z [36;1mmkdir -p  ./dependabot-job-1377112841-1779314845[0m
2026-05-20T22:07:32.2660777Z shell: /usr/bin/bash -e {0}
2026-05-20T22:07:32.2662127Z ##[endgroup]
2026-05-20T22:07:32.3010106Z ##[group]Run github/dependabot-action@main
2026-05-20T22:07:32.3011352Z env:
2026-05-20T22:07:32.3012161Z   DEPENDABOT_DISABLE_CLEANUP: 1
2026-05-20T22:07:32.3013509Z   DEPENDABOT_ENABLE_CONNECTIVITY_CHECK: 0
2026-05-20T22:07:32.3015267Z   GITHUB_TOKEN: ***
2026-05-20T22:07:32.3017006Z   GITHUB_DEPENDABOT_JOB_TOKEN: ***
2026-05-20T22:07:32.3018923Z   GITHUB_DEPENDABOT_CRED_TOKEN: ***
2026-05-20T22:07:32.3020042Z   GITHUB_REGISTRIES_PROXY: 
2026-05-20T22:07:32.3020989Z ##[endgroup]
2026-05-20T22:07:32.4612706Z 🤖 ~ starting update ~
2026-05-20T22:07:32.4638513Z Fetching job details
2026-05-20T22:07:32.8380822Z 🤖 ~ Failed to parse GITHUB_REGISTRIES_PROXY environment variable ~
2026-05-20T22:07:32.8385876Z ##[group]Pulling updater images
2026-05-20T22:07:32.8427356Z Pulling image ghcr.io/dependabot/dependabot-updater-nuget:4660a80db17f1781ad8b6cd01b52c279f9805b3f (attempt 1)...
2026-05-20T22:07:32.9574421Z Successfully sent metric (dependabot.action.ghcr_image_pull) to remote API endpoint
2026-05-20T22:08:25.2754515Z Pulled image ghcr.io/dependabot/dependabot-updater-nuget:4660a80db17f1781ad8b6cd01b52c279f9805b3f
2026-05-20T22:08:25.2766295Z Pulling image ghcr.io/dependabot/proxy:v2.0.20260427232629@sha256:4b22239a1bf296fb9a95264b395e965d41044bdb14ffd2caa95dc612fa9db64a (attempt 1)...
2026-05-20T22:08:25.4027168Z Successfully sent metric (dependabot.action.ghcr_image_pull) to remote API endpoint
2026-05-20T22:08:26.3211700Z Pulled image ghcr.io/dependabot/proxy:v2.0.20260427232629@sha256:4b22239a1bf296fb9a95264b395e965d41044bdb14ffd2caa95dc612fa9db64a
2026-05-20T22:08:26.3214260Z ##[endgroup]
2026-05-20T22:08:26.3214679Z Starting update process
2026-05-20T22:08:26.4808331Z Created proxy container: 69b707b5d3dbcc9b33f28f8ec1c8b242af380ec469272953fcd2912ea52f5cff
2026-05-20T22:08:26.6794261Z Created container: 1e6e460558212e7f4d6667e7aeee2a42c614a662fdb31b7f0b1d87f8ad53402c
2026-05-20T22:08:26.7585869Z   proxy | 2026/05/20 22:08:26 proxy starting, commit: 2803a0f0dec888dc2b556d1a9bea8a9d771f8167
2026-05-20T22:08:26.7588246Z   proxy | 2026/05/20 22:08:26 Listening (:1080)
2026-05-20T22:08:26.7806263Z Started container 1e6e460558212e7f4d6667e7aeee2a42c614a662fdb31b7f0b1d87f8ad53402c
2026-05-20T22:08:26.8283500Z updater | Updating certificates in /etc/ssl/certs...
2026-05-20T22:08:27.6257303Z updater | rehash: warning: skipping ca-certificates.crt,it does not contain exactly one certificate or CRL
2026-05-20T22:08:27.6317811Z updater | 1 added, 0 removed; done.
2026-05-20T22:08:27.6322703Z updater | Running hooks in /etc/ca-certificates/update.d...
2026-05-20T22:08:27.6335525Z updater | done.
2026-05-20T22:08:28.7020068Z updater | Job: {
2026-05-20T22:08:28.7020484Z   "command": "version",
2026-05-20T22:08:28.7020825Z   "allowed-updates": [
2026-05-20T22:08:28.7021136Z     {
2026-05-20T22:08:28.7021424Z       "dependency-type": "all"
2026-05-20T22:08:28.7021764Z     }
2026-05-20T22:08:28.7022006Z   ],
2026-05-20T22:08:28.7022286Z   "commit-message-options": {
2026-05-20T22:08:28.7022734Z     "prefix": null,
2026-05-20T22:08:28.7023075Z     "prefix-development": null,
2026-05-20T22:08:28.7023638Z     "include-scope": null
2026-05-20T22:08:28.7023961Z   },
2026-05-20T22:08:28.7024234Z   "credentials-metadata": [
2026-05-20T22:08:28.7024826Z     {
2026-05-20T22:08:28.7025124Z       "type": "git_source",
2026-05-20T22:08:28.7025455Z       "host": "github.com"
2026-05-20T22:08:28.7025764Z     }
2026-05-20T22:08:28.7026007Z   ],
2026-05-20T22:08:28.7026264Z   "debug": null,
2026-05-20T22:08:28.7026568Z   "dependencies": null,
2026-05-20T22:08:28.7026817Z   "dependency-groups": [
2026-05-20T22:08:28.7027026Z     {
2026-05-20T22:08:28.7027209Z       "name": "asp",
2026-05-20T22:08:28.7027516Z       "rules": {
2026-05-20T22:08:28.7027808Z         "patterns": [
2026-05-20T22:08:28.7028164Z           "Asp.Versioning.*"
2026-05-20T22:08:28.7028545Z         ]
2026-05-20T22:08:28.7028800Z       }
2026-05-20T22:08:28.7029017Z     },
2026-05-20T22:08:28.7029427Z     {
2026-05-20T22:08:28.7030169Z       "name": "microsoft",
2026-05-20T22:08:28.7030493Z       "rules": {
2026-05-20T22:08:28.7030774Z         "patterns": [
2026-05-20T22:08:28.7031116Z           "Microsoft.AspNetCore.*",
2026-05-20T22:08:28.7031572Z           "Microsoft.Extensions.*"
2026-05-20T22:08:28.7031928Z         ],
2026-05-20T22:08:28.7032199Z         "exclude-patterns": [
2026-05-20T22:08:28.7032679Z           "Microsoft.Extensions.Logging.ApplicationInsights"
2026-05-20T22:08:28.7033171Z         ]
2026-05-20T22:08:28.7033605Z       }
2026-05-20T22:08:28.7033764Z     },
2026-05-20T22:08:28.7033932Z     {
2026-05-20T22:08:28.7034105Z       "name": "serilog",
2026-05-20T22:08:28.7034315Z       "rules": {
2026-05-20T22:08:28.7034491Z         "patterns": [
2026-05-20T22:08:28.7034689Z           "Serilog*"
2026-05-20T22:08:28.7034874Z         ]
2026-05-20T22:08:28.7035040Z       }
2026-05-20T22:08:28.7035210Z     }
2026-05-20T22:08:28.7035369Z   ],
2026-05-20T22:08:28.7035565Z   "dependency-group-to-refresh": null,
2026-05-20T22:08:28.7035943Z   "existing-pull-requests": [
2026-05-20T22:08:28.7036170Z     {
2026-05-20T22:08:28.7036345Z       "pr-number": 136,
2026-05-20T22:08:28.7036558Z       "dependencies": [
2026-05-20T22:08:28.7036744Z         {
2026-05-20T22:08:28.7036952Z           "dependency-name": "FluentAssertions",
2026-05-20T22:08:28.7037245Z           "dependency-version": "8.0.0",
2026-05-20T22:08:28.7037500Z           "directory": "/"
2026-05-20T22:08:28.7037697Z         }
2026-05-20T22:08:28.7037862Z       ]
2026-05-20T22:08:28.7038024Z     },
2026-05-20T22:08:28.7038186Z     {
2026-05-20T22:08:28.7038347Z       "pr-number": 154,
2026-05-20T22:08:28.7038552Z       "dependencies": [
2026-05-20T22:08:28.7038758Z         {
2026-05-20T22:08:28.7039006Z           "dependency-name": "Microsoft.Extensions.Logging",
2026-05-20T22:08:28.7039320Z           "dependency-version": "9.0.4",
2026-05-20T22:08:28.7039584Z           "directory": "/"
2026-05-20T22:08:28.7039797Z         }
2026-05-20T22:08:28.7040184Z       ]
2026-05-20T22:08:28.7040344Z     },
2026-05-20T22:08:28.7040521Z     {
2026-05-20T22:08:28.7040700Z       "pr-number": 240,
2026-05-20T22:08:28.7040920Z       "dependencies": [
2026-05-20T22:08:28.7041104Z         {
2026-05-20T22:08:28.7041308Z           "dependency-name": "Scalar.AspNetCore",
2026-05-20T22:08:28.7041590Z           "dependency-version": "2.2.3"
2026-05-20T22:08:28.7041879Z         }
2026-05-20T22:08:28.7042036Z       ]
2026-05-20T22:08:28.7042207Z     },
2026-05-20T22:08:28.7042369Z     {
2026-05-20T22:08:28.7042541Z       "pr-number": 289,
2026-05-20T22:08:28.7042744Z       "dependencies": [
2026-05-20T22:08:28.7042942Z         {
2026-05-20T22:08:28.7043556Z           "dependency-name": "Microsoft.Playwright",
2026-05-20T22:08:28.7043984Z           "dependency-version": "1.58.0"
2026-05-20T22:08:28.7044231Z         }
2026-05-20T22:08:28.7044392Z       ]
2026-05-20T22:08:28.7044563Z     },
2026-05-20T22:08:28.7044742Z     {
2026-05-20T22:08:28.7044923Z       "pr-number": 371,
2026-05-20T22:08:28.7045150Z       "dependencies": [
2026-05-20T22:08:28.7045357Z         {
2026-05-20T22:08:28.7045583Z           "dependency-name": "Microsoft.NET.Test.Sdk",
2026-05-20T22:08:28.7045888Z           "dependency-version": "18.4.0"
2026-05-20T22:08:28.7046123Z         }
2026-05-20T22:08:28.7046301Z       ]
2026-05-20T22:08:28.7046472Z     },
2026-05-20T22:08:28.7046632Z     {
2026-05-20T22:08:28.7046807Z       "pr-number": 129,
2026-05-20T22:08:28.7047006Z       "dependencies": [
2026-05-20T22:08:28.7047199Z         {
2026-05-20T22:08:28.7047384Z           "dependency-name": "Serilog",
2026-05-20T22:08:28.7047643Z           "dependency-version": "4.2.0",
2026-05-20T22:08:28.7047883Z           "directory": "/"
2026-05-20T22:08:28.7048089Z         },
2026-05-20T22:08:28.7048260Z         {
2026-05-20T22:08:28.7048503Z           "dependency-name": "Serilog.Settings.Configuration",
2026-05-20T22:08:28.7048817Z           "dependency-version": "9.0.0",
2026-05-20T22:08:28.7049061Z           "directory": "/"
2026-05-20T22:08:28.7049269Z         }
2026-05-20T22:08:28.7049599Z       ]
2026-05-20T22:08:28.7049764Z     },
2026-05-20T22:08:28.7049926Z     {
2026-05-20T22:08:28.7050104Z       "pr-number": 170,
2026-05-20T22:08:28.7050298Z       "dependencies": [
2026-05-20T22:08:28.7050489Z         {
2026-05-20T22:08:28.7050700Z           "dependency-name": "Scalar.AspNetCore",
2026-05-20T22:08:28.7051021Z           "dependency-version": "2.2.5"
2026-05-20T22:08:28.7051261Z         },
2026-05-20T22:08:28.7051427Z         {
2026-05-20T22:08:28.7051645Z           "dependency-name": "SonarAnalyzer.CSharp",
2026-05-20T22:08:28.7051955Z           "dependency-version": "10.9.0.115408"
2026-05-20T22:08:28.7052205Z         },
2026-05-20T22:08:28.7052366Z         {
2026-05-20T22:08:28.7052571Z           "dependency-name": "SonarAnalyzer.CSharp",
2026-05-20T22:08:28.7052861Z           "dependency-version": "10.9.0.115408"
2026-05-20T22:08:28.7053102Z         },
2026-05-20T22:08:28.7053480Z         {
2026-05-20T22:08:28.7053789Z           "dependency-name": "SonarAnalyzer.CSharp",
2026-05-20T22:08:28.7054096Z           "dependency-version": "10.9.0.115408"
2026-05-20T22:08:28.7054342Z         },
2026-05-20T22:08:28.7054503Z         {
2026-05-20T22:08:28.7054706Z           "dependency-name": "SonarAnalyzer.CSharp",
2026-05-20T22:08:28.7054996Z           "dependency-version": "10.9.0.115408"
2026-05-20T22:08:28.7055252Z         }
2026-05-20T22:08:28.7055416Z       ]
2026-05-20T22:08:28.7055587Z     },
2026-05-20T22:08:28.7055755Z     {
2026-05-20T22:08:28.7055935Z       "pr-number": 174,
2026-05-20T22:08:28.7056142Z       "dependencies": [
2026-05-20T22:08:28.7056344Z         {
2026-05-20T22:08:28.7056564Z           "dependency-name": "Microsoft.NET.Test.Sdk",
2026-05-20T22:08:28.7056862Z           "dependency-version": "17.14.0"
2026-05-20T22:08:28.7057104Z         },
2026-05-20T22:08:28.7057262Z         {
2026-05-20T22:08:28.7057469Z           "dependency-name": "Microsoft.NET.Test.Sdk",
2026-05-20T22:08:28.7057921Z           "dependency-version": "17.14.0"
2026-05-20T22:08:28.7058172Z         },
2026-05-20T22:08:28.7058335Z         {
2026-05-20T22:08:28.7058522Z           "dependency-name": "Serilog",
2026-05-20T22:08:28.7058782Z           "dependency-version": "4.3.0"
2026-05-20T22:08:28.7059014Z         }
2026-05-20T22:08:28.7059172Z       ]
2026-05-20T22:08:28.7059330Z     }
2026-05-20T22:08:28.7059495Z   ],
2026-05-20T22:08:28.7059700Z   "existing-group-pull-requests": [
2026-05-20T22:08:28.7059935Z     {
2026-05-20T22:08:28.7060127Z       "dependency-group-name": "serilog",
2026-05-20T22:08:28.7060408Z       "dependencies": [
2026-05-20T22:08:28.7060606Z         {
2026-05-20T22:08:28.7060858Z           "dependency-name": "Serilog.Sinks.ApplicationInsights",
2026-05-20T22:08:28.7061194Z           "dependency-version": "5.0.1"
2026-05-20T22:08:28.7061437Z         }
2026-05-20T22:08:28.7061600Z       ],
2026-05-20T22:08:28.7061768Z       "pr-number": 386
2026-05-20T22:08:28.7061963Z     }
2026-05-20T22:08:28.7062136Z   ],
2026-05-20T22:08:28.7062311Z   "experiments": {
2026-05-20T22:08:28.7062535Z     "record-ecosystem-versions": true,
2026-05-20T22:08:28.7062823Z     "record-update-job-unknown-error": true,
2026-05-20T22:08:28.7063093Z     "proxy-cached": true,
2026-05-20T22:08:28.7063588Z     "enable-corepack-for-npm-and-yarn": true,
2026-05-20T22:08:28.7063914Z     "enable-private-registry-for-corepack": true,
2026-05-20T22:08:28.7064260Z     "allow-refresh-for-existing-pr-dependencies": true,
2026-05-20T22:08:28.7064617Z     "allow-refresh-group-with-all-dependencies": true,
2026-05-20T22:08:28.7064930Z     "azure-registry-backup": true,
2026-05-20T22:08:28.7065226Z     "enable-enhanced-error-details-for-updater": true,
2026-05-20T22:08:28.7065522Z     "gradle-lockfile-updater": true,
2026-05-20T22:08:28.7065860Z     "enable-exclude-paths-subdirectory-manifest-files": true,
2026-05-20T22:08:28.7066206Z     "enable-audit-fix-fallback": true
2026-05-20T22:08:28.7066464Z   },
2026-05-20T22:08:28.7066644Z   "ignore-conditions": [
2026-05-20T22:08:28.7066853Z     {
2026-05-20T22:08:28.7067212Z       "dependency-name": "FluentAssertions",
2026-05-20T22:08:28.7067496Z       "version-requirement": ">= 8.a, < 9",
2026-05-20T22:08:28.7067783Z       "update-types": null,
2026-05-20T22:08:28.7068027Z       "source": "@dependabot ignore command",
2026-05-20T22:08:28.7068298Z       "updated-at": "2025-01-16T19:25:27Z"
2026-05-20T22:08:28.7068537Z     },
2026-05-20T22:08:28.7068705Z     {
2026-05-20T22:08:28.7068909Z       "dependency-name": "Microsoft.Playwright",
2026-05-20T22:08:28.7069197Z       "version-requirement": ">= 0",
2026-05-20T22:08:28.7069446Z       "update-types": null,
2026-05-20T22:08:28.7069680Z       "source": ".github/dependabot.yml",
2026-05-20T22:08:28.7069935Z       "updated-at": "2026-02-03T06:03:27Z"
2026-05-20T22:08:28.7070170Z     },
2026-05-20T22:08:28.7070344Z     {
2026-05-20T22:08:28.7070537Z       "dependency-name": "Scalar.AspNetCore",
2026-05-20T22:08:28.7070797Z       "version-requirement": "> 2.2.3",
2026-05-20T22:08:28.7071062Z       "update-types": null,
2026-05-20T22:08:28.7071320Z       "source": "@dependabot ignore command",
2026-05-20T22:08:28.7071595Z       "updated-at": "2025-05-04T07:46:32Z"
2026-05-20T22:08:28.7071835Z     },
2026-05-20T22:08:28.7071997Z     {
2026-05-20T22:08:28.7072205Z       "dependency-name": "SonarAnalyzer.CSharp",
2026-05-20T22:08:28.7072505Z       "version-requirement": "> 10.8.0.113526",
2026-05-20T22:08:28.7072774Z       "update-types": null,
2026-05-20T22:08:28.7073009Z       "source": "@dependabot ignore command",
2026-05-20T22:08:28.7073415Z       "updated-at": "2025-05-04T07:46:32Z"
2026-05-20T22:08:28.7073768Z     }
2026-05-20T22:08:28.7073936Z   ],
2026-05-20T22:08:28.7074128Z   "lockfile-only": false,
2026-05-20T22:08:28.7074363Z   "max-updater-run-time": 2700,
2026-05-20T22:08:28.7074615Z   "package-manager": "nuget",
2026-05-20T22:08:28.7074872Z   "requirements-update-strategy": null,
2026-05-20T22:08:28.7075143Z   "reject-external-code": false,
2026-05-20T22:08:28.7075544Z   "security-advisories": [],
2026-05-20T22:08:28.7075794Z   "security-updates-only": false,
2026-05-20T22:08:28.7076034Z   "source": {
2026-05-20T22:08:28.7076221Z     "provider": "github",
2026-05-20T22:08:28.7076458Z     "repo": "Tsingis/bitcoin-web-api",
2026-05-20T22:08:28.7076712Z     "branch": null,
2026-05-20T22:08:28.7076913Z     "directory": "/.",
2026-05-20T22:08:28.7077350Z     "api-endpoint": "https://api.github.com/",
2026-05-20T22:08:28.7077663Z     "hostname": "github.com"
2026-05-20T22:08:28.7077888Z   },
2026-05-20T22:08:28.7078077Z   "updating-a-pull-request": false,
2026-05-20T22:08:28.7078343Z   "update-subdependencies": true,
2026-05-20T22:08:28.7078597Z   "vendor-dependencies": false,
2026-05-20T22:08:28.7078839Z   "enable-beta-ecosystems": false,
2026-05-20T22:08:28.7079082Z   "repo-private": false,
2026-05-20T22:08:28.7079300Z   "cooldown": {
2026-05-20T22:08:28.7079491Z     "default-days": 1,
2026-05-20T22:08:28.7079709Z     "semver-major-days": 0,
2026-05-20T22:08:28.7079935Z     "semver-minor-days": 0,
2026-05-20T22:08:28.7080164Z     "semver-patch-days": 0,
2026-05-20T22:08:28.7080377Z     "include": [],
2026-05-20T22:08:28.7080569Z     "exclude": [
2026-05-20T22:08:28.7080758Z       "Microsoft.*"
2026-05-20T22:08:28.7080971Z     ]
2026-05-20T22:08:28.7081139Z   },
2026-05-20T22:08:28.7081329Z   "multi-ecosystem-update": false,
2026-05-20T22:08:28.7081569Z   "exclude-paths": null
2026-05-20T22:08:28.7081781Z }
2026-05-20T22:08:29.0259708Z updater | 2026/05/20 22:08:29 INFO Running command: git clone --no-tags --depth 1 --recurse-submodules --shallow-submodules https://github.com/Tsingis/bitcoin-web-api /home/dependabot/dependabot-updater/repo
2026-05-20T22:08:29.2396938Z   proxy | 2026/05/20 22:08:29 [002] GET https://github.com:443/Tsingis/bitcoin-web-api/info/refs?service=git-upload-pack
2026-05-20T22:08:29.2397915Z 2026/05/20 22:08:29 [002] * authenticating git server request (host: github.com)
2026-05-20T22:08:29.3910196Z   proxy | 2026/05/20 22:08:29 [002] 200 https://github.com:443/Tsingis/bitcoin-web-api/info/refs?service=git-upload-pack
2026-05-20T22:08:29.4265282Z   proxy | 2026/05/20 22:08:29 [004] POST https://github.com:443/Tsingis/bitcoin-web-api/git-upload-pack
2026-05-20T22:08:29.4265906Z 2026/05/20 22:08:29 [004] * authenticating git server request (host: github.com)
2026-05-20T22:08:29.5104895Z   proxy | 2026/05/20 22:08:29 [004] 200 https://github.com:443/Tsingis/bitcoin-web-api/git-upload-pack
2026-05-20T22:08:29.5480616Z   proxy | 2026/05/20 22:08:29 [006] POST https://github.com:443/Tsingis/bitcoin-web-api/git-upload-pack
2026-05-20T22:08:29.5481486Z 2026/05/20 22:08:29 [006] * authenticating git server request (host: github.com)
2026-05-20T22:08:29.6635040Z   proxy | 2026/05/20 22:08:29 [006] 200 https://github.com:443/Tsingis/bitcoin-web-api/git-upload-pack
2026-05-20T22:08:29.8234611Z updater | Renaming '/home/dependabot/dependabot-updater/repo/nuget.config' to '/home/dependabot/dependabot-updater/repo/NuGet.Config'
2026-05-20T22:08:29.8609646Z updater | Currently installed SDKs: 8.0.418 9.0.311 10.0.103
2026-05-20T22:08:29.8834599Z updater | Discovering global.json files for SDK install with repo root '/home/dependabot/dependabot-updater/repo' and update directories '/.'
2026-05-20T22:08:29.9192399Z updater |   Processing update directory '/.' with regex '^/home/dependabot/dependabot-updater/repo/\.$'
2026-05-20T22:08:29.9267194Z updater | Discovered global.json files: 
2026-05-20T22:08:29.9394708Z updater | 8.0.418 [/usr/local/dotnet/current/sdk]
2026-05-20T22:08:29.9395194Z 9.0.311 [/usr/local/dotnet/current/sdk]
2026-05-20T22:08:29.9395598Z 10.0.103 [/usr/local/dotnet/current/sdk]
2026-05-20T22:08:29.9534461Z updater | User-level NuGet.Config set to ...
2026-05-20T22:08:29.9568604Z updater | <?xml version="1.0" encoding="utf-8"?>
2026-05-20T22:08:29.9569197Z updater | <configuration>
2026-05-20T22:08:29.9572603Z updater |   <packageSources>
2026-05-20T22:08:29.9573659Z     <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
2026-05-20T22:08:29.9574615Z   </packageSources>
2026-05-20T22:08:29.9574953Z updater | </configuration>
2026-05-20T22:08:29.9714418Z updater | Base commit SHA: 44c8b43023f70742fbd13c640f6b415817b305da
2026-05-20T22:08:30.2824274Z updater | 2026/05/20 22:08:30 INFO Starting update job of type group_update_all_versions
2026-05-20T22:08:30.4820547Z   proxy | 2026/05/20 22:08:30 [008] POST /update_jobs/1377112841/increment_metric
2026-05-20T22:08:30.6119109Z   proxy | 2026/05/20 22:08:30 [008] 204 /update_jobs/1377112841/increment_metric
2026-05-20T22:08:30.6316472Z updater | 2026/05/20 22:08:30 INFO Starting update for group asp
2026-05-20T22:08:30.6499125Z updater | 2026/05/20 22:08:30 INFO Temporarily removing `global.json` from `/home/dependabot/dependabot-updater/repo`.
2026-05-20T22:08:30.6708334Z updater | 2026/05/20 22:08:30 INFO Restoring `global.json` to `/home/dependabot/dependabot-updater/repo`.
2026-05-20T22:08:30.6711020Z updater | 2026/05/20 22:08:30 INFO Discovering build files in workspace [/home/dependabot/dependabot-updater/repo].
2026-05-20T22:08:30.6721814Z updater | 2026/05/20 22:08:30 INFO   Discovered [.config/dotnet-tools.json] file.
2026-05-20T22:08:30.6773732Z updater | 2026/05/20 22:08:30 INFO   Discovered [global.json] file.
2026-05-20T22:08:30.6819938Z updater | 2026/05/20 22:08:30 INFO   Discovering projects beneath [.].
2026-05-20T22:08:30.6826362Z updater | 2026/05/20 22:08:30 INFO     Entry points found: /home/dependabot/dependabot-updater/repo/app.slnx
2026-05-20T22:08:30.6850225Z updater | 2026/05/20 22:08:30 INFO     Expanding solution: /home/dependabot/dependabot-updater/repo/app.slnx:
2026-05-20T22:08:30.7258371Z updater | 2026/05/20 22:08:30 INFO       Expanded project: /home/dependabot/dependabot-updater/repo/src/Api/Api.csproj
2026-05-20T22:08:30.7284805Z updater | 2026/05/20 22:08:30 INFO       Expanded project: /home/dependabot/dependabot-updater/repo/src/Common/Common.csproj
2026-05-20T22:08:30.7314404Z updater | 2026/05/20 22:08:30 INFO       Expanded project: /home/dependabot/dependabot-updater/repo/src/Services/Services.csproj
2026-05-20T22:08:30.7344977Z updater | 2026/05/20 22:08:30 INFO       Expanded project: /home/dependabot/dependabot-updater/repo/tests/ArchitectureTests/ArchitectureTests.csproj
2026-05-20T22:08:30.7347304Z updater | 2026/05/20 22:08:30 INFO       Expanded project: /home/dependabot/dependabot-updater/repo/tests/E2ETests/E2ETests.csproj
2026-05-20T22:08:30.7374642Z 2026/05/20 22:08:30 INFO       Expanded project: /home/dependabot/dependabot-updater/repo/tests/IntegrationTests/IntegrationTests.csproj
2026-05-20T22:08:30.7385255Z 2026/05/20 22:08:30 INFO       Expanded project: /home/dependabot/dependabot-updater/repo/tests/UnitTests/UnitTests.csproj
2026-05-20T22:08:30.7934114Z updater | 2026/05/20 22:08:30 INFO   No packages.config file found.
2026-05-20T22:08:30.7937522Z updater | 2026/05/20 22:08:30 INFO   No packages.config file found.
2026-05-20T22:08:30.7938947Z updater | 2026/05/20 22:08:30 INFO   No packages.config file found.
2026-05-20T22:08:30.7939970Z updater | 2026/05/20 22:08:30 INFO   No packages.config file found.
2026-05-20T22:08:30.7942822Z updater | 2026/05/20 22:08:30 INFO   No packages.config file found.
2026-05-20T22:08:30.7951671Z updater | 2026/05/20 22:08:30 INFO   No packages.config file found.
2026-05-20T22:08:30.7955189Z updater | 2026/05/20 22:08:30 INFO   No packages.config file found.
2026-05-20T22:08:30.8095772Z updater | 2026/05/20 22:08:30 WARN Unable to determine property 'TargetFrameworks' for project [/home/dependabot/dependabot-updater/repo/src/Api/Api.csproj]:
2026-05-20T22:08:30.8097966Z STDOUT:
2026-05-20T22:08:30.8106486Z 8.0.418 [/usr/local/dotnet/current/sdk]
2026-05-20T22:08:30.8107415Z 9.0.311 [/usr/local/dotnet/current/sdk]
2026-05-20T22:08:30.8117421Z 10.0.103 [/usr/localupdater | /dotnet/current/sdk]
2026-05-20T22:08:30.8118124Z 
2026-05-20T22:08:30.8118289Z 
2026-05-20T22:08:30.8118795Z STDERR:
2026-05-20T22:08:30.8119457Z The command could not be loaded, possibly because:
2026-05-20T22:08:30.8120586Z   * You intended to execute a .NET application:
2026-05-20T22:08:30.8122065Z       The application 'msbuild' does not exist or is not a managed .dll or .exe.
2026-05-20T22:08:30.8123455Z   * You intended to execute a .NET SDK command:
2026-05-20T22:08:30.8124361Z       A compatible .NET SDK was not found.
2026-05-20T22:08:30.8124853Z 
2026-05-20T22:08:30.8125375Z Requested SDK version: 10.0.300
2026-05-20T22:08:30.8126307Z global.json file: /home/dependabot/dependabot-updater/repo/global.json
2026-05-20T22:08:30.8126823Z 
2026-05-20T22:08:30.8127051Z Installed SDKs:
2026-05-20T22:08:30.8127266Z 
2026-05-20T22:08:30.8127920Z Install the [10.0.300] .NET SDK or update [/home/dependabot/dependabot-updater/repo/global.json] to match an installed SDK.
2026-05-20T22:08:30.8128650Z 
2026-05-20T22:08:30.8129026Z Learn about SDK resolution:
2026-05-20T22:08:30.8129598Z https://aka.ms/dotnet/sdk-not-found
2026-05-20T22:08:30.8129992Z 
2026-05-20T22:08:30.8129997Z 
2026-05-20T22:08:30.8130003Z 
2026-05-20T22:08:30.8130749Z updater | 2026/05/20 22:08:30 INFO Performing single restore for project /home/dependabot/dependabot-updater/repo/src/Api/Api.csproj
2026-05-20T22:08:30.8166334Z updater | 2026/05/20 22:08:30 WARN Unable to determine property 'NETCoreSdkVersion' for project [/home/dependabot/dependabot-updater/repo/src/Api/Api.csproj]:
2026-05-20T22:08:30.8167651Z STDOUT:
2026-05-20T22:08:30.8168406Z 8.0.418 [/usr/local/dotnet/current/sdk]
2026-05-20T22:08:30.8169294Z 9.0.311 [/usr/local/dotnet/current/sdk]
2026-05-20T22:08:30.8170207Z 10.0.103 [/usr/locaupdater | l/dotnet/current/sdk]
2026-05-20T22:08:30.8170727Z 
2026-05-20T22:08:30.8170882Z 
2026-05-20T22:08:30.8171368Z STDERR:
2026-05-20T22:08:30.8172000Z The command could not be loaded, possibly because:
2026-05-20T22:08:30.8173122Z   * You intended to execute a .NET application:
2026-05-20T22:08:30.8174543Z       The application 'msbuild' does not exist or is not a managed .dll or .exe.
2026-05-20T22:08:30.8175699Z   * You intended to execute a .NET SDK commaupdater | nd:
2026-05-20T22:08:30.8176586Z       A compatible .NET SDK was not found.
2026-05-20T22:08:30.8177556Z 
2026-05-20T22:08:30.8178026Z Requested SDK version: 10.0.300
2026-05-20T22:08:30.8178854Z global.json file: /home/dependabot/dependabot-updater/repo/global.json
2026-05-20T22:08:30.8179591Z 
2026-05-20T22:08:30.8179887Z Installed SDKs:
2026-05-20T22:08:30.8180519Z 
2026-05-20T22:08:30.8181343Z Install the [10.0.300] .NET SDK or update [/home/dependabot/dependabot-updater/repo/gloupdater | bal.json] to match an installed SDK.
2026-05-20T22:08:30.8182388Z 
2026-05-20T22:08:30.8182723Z Learn about SDK resolution:
2026-05-20T22:08:30.8183746Z https://aka.ms/dotnet/sdk-not-found
2026-05-20T22:08:30.8184259Z 
2026-05-20T22:08:30.8184414Z 
2026-05-20T22:08:30.8184630Z 
2026-05-20T22:08:30.8221595Z updater | 2026/05/20 22:08:30 WARN Unable to determine targets for project [/home/dependabot/dependabot-updater/repo/src/Api/Api.csproj]:
2026-05-20T22:08:30.8222639Z STDOUT:
2026-05-20T22:08:30.8224841Z 8.0.418 [/usr/local/dotnet/current/sdk]
2026-05-20T22:08:30.8225450Z 9.0.311 [/usr/local/dotnet/current/sdk]
2026-05-20T22:08:30.8226031Z 10.0.103 [/usr/local/dotnet/current/sdk]
2026-05-20T22:08:30.8226521Z 
2026-05-20T22:08:30.8226527Z 
2026-05-20T22:08:30.8226713Z STDERR:
2026-05-20T22:08:30.8227222Z The command could not be loaded, possibly because:
2026-05-20T22:08:30.8227896Z   * You intended to execute a .NET application:
2026-05-20T22:08:30.8228712Z       The application 'msbuild' does not exist or is not a managed .dll or .exe.
2026-05-20T22:08:30.8229444Z   * You intended to execute a .NET SDK command:
2026-05-20T22:08:30.8230024Z       A compatible .NET SDK was not found.
2026-05-20T22:08:30.8230473Z 
2026-05-20T22:08:30.8230750Z Requested SDK version: 10.0.300
2026-05-20T22:08:30.8231426Z global.json file: /home/dependabot/dependabot-updater/repo/global.json
2026-05-20T22:08:30.8231964Z 
2026-05-20T22:08:30.8232099Z Installed SDKs:
2026-05-20T22:08:30.8232478Z 
2026-05-20T22:08:30.8233093Z Install the [10.0.300] .NET SDK or update [/home/dependabot/dependabot-updater/repo/global.json] to match an installed SDK.
2026-05-20T22:08:30.8234393Z 
2026-05-20T22:08:30.8234577Z Learn about SDK resolution:
2026-05-20T22:08:30.8235171Z https://aka.ms/dotnet/sdk-not-found
2026-05-20T22:08:30.8235513Z 
2026-05-20T22:08:30.8235519Z 
2026-05-20T22:08:30.8235524Z 
2026-05-20T22:08:30.8296326Z updater | 2026/05/20 22:08:30 WARN   Error determining dependencies from `/home/dependabot/dependabot-updater/repo/src/Api/Api.csproj`:
2026-05-20T22:08:30.8297851Z STDOUT:
2026-05-20T22:08:30.8299043Z 8.0.418 [/usr/local/dotnet/current/sdk]
2026-05-20T22:08:30.8300075Z 9.0.311 [/usr/local/dotnet/current/sdk]
2026-05-20T22:08:30.8301331Z 10.0.103 [/usr/local/dotnet/current/sdk]
2026-05-20T22:08:30.8302282Z 
2026-05-20T22:08:30.8312723Z updater | 
2026-05-20T22:08:30.8313959Z STDERR:
2026-05-20T22:08:30.8314870Z The command could not be loaded, possibly because:
2026-05-20T22:08:30.8315828Z   * You intended to execute a .NET application:
2026-05-20T22:08:30.8316832Z       The application 'msbuild' does not exist or is not a managed .dll or .exe.
2026-05-20T22:08:30.8317944Z   * You intended to execute a .NET SDK command:
2026-05-20T22:08:30.8318831Z       A compatible .NET SDK was not found.
2026-05-20T22:08:30.8319449Z 
2026-05-20T22:08:30.8319842Z Requested SDK version: 10.0.300
2026-05-20T22:08:30.8320900Z global.json file: /home/dependabot/dependabot-updater/repo/global.json
2026-05-20T22:08:30.8321610Z 
2026-05-20T22:08:30.8322047Z Installed SDKs:
2026-05-20T22:08:30.8322505Z 
2026-05-20T22:08:30.8323590Z Install the [10.0.300] .NET SDK or update [/home/dependabot/dependabot-updater/repo/global.json] to match an installed SDK.
2026-05-20T22:08:30.8324777Z 
2026-05-20T22:08:30.8325273Z Learn about SDK resolution:
2026-05-20T22:08:30.8325978Z https://aka.ms/dotnet/sdk-not-found
2026-05-20T22:08:30.8326599Z 
2026-05-20T22:08:30.8326778Z 
2026-05-20T22:08:30.8524880Z updater | 2026/05/20 22:08:30 ERROR Error type: unknown_error
2026-05-20T22:08:30.8527231Z - error-class: FileNotFoundException
2026-05-20T22:08:30.8529187Z - error-message: System.IO.FileNotFoundException: Could not find file '/tmp/msbuild_1393d57d-4d85-49cc-ada7-891d7e26552d.binlog'.
2026-05-20T22:08:30.8530961Z File name: '/tmp/msbuild_1393d57d-4d85-49cc-ada7-891d7e26552d.binlog'
2026-05-20T22:08:30.8532319Z    at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirError)
2026-05-20T22:08:30.8534534Z    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode, Boolean failForSymlink, Boolean& wasSymlink, Func`4 createOpenException)
2026-05-20T22:08:30.8538855Z    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, UnixFileMode openPermissions, Int64& fileLength, UnixFileMode& filePermissions, Boolean failForSymlink, Boolean& wasSymlink, Func`4 createOpenException)
2026-05-20T22:08:30.8542491Z    at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
2026-05-20T22:08:30.8544704Z    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
2026-05-20T22:08:30.8546952Z    at Microsoft.Build.Logging.StructuredLogger.BinaryLog.ReadBuild(String filePath, Progress progress, ReaderSettings readerSettings) in C:\MSBuildStructuredLog\src\StructuredLogger\BinaryLog.cs:line 37
2026-05-20T22:08:30.8549594Z    at Microsoft.Build.Logging.StructuredLogger.BinaryLog.ReadBuild(String filePath, Progress progress) in C:\MSBuildStructuredLog\src\StructuredLogger\BinaryLog.cs:line 31
2026-05-20T22:08:30.8551869Z    at Microsoft.Build.Logging.StructuredLogger.BinaryLog.ReadBuild(String filePath) in C:\MSBuildStructuredLog\src\StructuredLogger\BinaryLog.cs:line 30
2026-05-20T22:08:30.8555320Z    at NuGetUpdater.Core.Discover.SdkProjectDiscovery.DiscoverAsync(String repoRootPath, String workspacePath, String startingProjectPath, ExperimentsManager experimentsManager, ILogger logger) in /opt/nuget/lib/NuGetUpdater/NuGetUpdater.Core/Discover/SdkProjectDiscovery.cs:line 216
2026-05-20T22:08:30.8559423Z    at NuGetUpdater.Core.Discover.DiscoveryWorker.RunForProjectPathsAsync(String repoRootPath, String workspacePath, IEnumerable`1 projectPaths) in /opt/nuget/lib/NuGetUpdater/NuGetUpdater.Core/Discover/DiscoveryWorker.cs:line 376
2026-05-20T22:08:30.8562595Z    at NuGetUpdater.Core.Discover.DiscoveryWorker.RunForDirectoryAsync(String repoRootPath, String workspacePath) in /opt/nuget/lib/NuGetUpdater/NuGetUpdater.Core/Discover/DiscoveryWorker.cs:line 201
2026-05-20T22:08:30.8565678Z    at NuGetUpdater.Core.Discover.DiscoveryWorker.RunAsync(String repoRootPath, String workspacePath) in /opt/nuget/lib/NuGetUpdater/NuGetUpdater.Core/Discover/DiscoveryWorker.cs:line 102
2026-05-20T22:08:30.8570523Z    at NuGetUpdater.Core.Run.UpdateHandlers.GroupUpdateAllVersionsHandler.RunGroupedDependencyUpdates(Job job, DirectoryInfo originalRepoContentsPath, DirectoryInfo caseInsensitiveRepoContentsPath, String baseCommitSha, IDiscoveryWorker discoveryWorker, IAnalyzeWorker analyzeWorker, IUpdaterWorker updaterWorker, IApiHandler apiHandler, ExperimentsManager experimentsManager, ILogger logger) in /opt/nuget/lib/NuGetUpdater/NuGetUpdater.Core/Run/UpdateHandlers/GroupUpdateAllVersionsHandler.cs:line 65
2026-05-20T22:08:30.8579729Z    at NuGetUpdater.Core.Run.UpdateHandlers.GroupUpdateAllVersionsHandler.HandleAsync(Job job, DirectoryInfo originalRepoContentsPath, DirectoryInfo caseInsensitiveRepoContentsPath, String baseCommitSha, IDiscoveryWorker discoveryWorker, IAnalyzeWorker analyzeWorker, IUpdaterWorker updaterWorker, IApiHandler apiHandler, ExperimentsManager experimentsManager, ILogger logger) in /opt/nuget/lib/NuGetUpdater/NuGetUpdater.Core/Run/UpdateHandlers/GroupUpdateAllVersionsHandler.cs:line 48
2026-05-20T22:08:30.8585230Z    at NuGetUpdater.Core.Run.RunWorker.RunScenarioHandlersWithErrorHandlingAsync(Job job, DirectoryInfo repoContentsPath, DirectoryInfo caseInsensitiveRepoContentsPath, String baseCommitSha, ExperimentsManager experimentsManager) in /opt/nuget/lib/NuGetUpdater/NuGetUpdater.Core/Run/RunWorker.cs:line 100
2026-05-20T22:08:30.8587549Z - package-manager: nuget
2026-05-20T22:08:30.8587977Z - job-id: 1377112841
2026-05-20T22:08:30.8632937Z   proxy | 2026/05/20 22:08:30 [010] POST /update_jobs/1377112841/record_update_job_error
2026-05-20T22:08:30.9399576Z   proxy | 2026/05/20 22:08:30 [010] 204 /update_jobs/1377112841/record_update_job_error
2026-05-20T22:08:30.9462455Z   proxy | 2026/05/20 22:08:30 [012] POST /update_jobs/1377112841/record_update_job_unknown_error
2026-05-20T22:08:31.0121835Z   proxy | 2026/05/20 22:08:31 [012] 204 /update_jobs/1377112841/record_update_job_unknown_error
2026-05-20T22:08:31.0175024Z   proxy | 2026/05/20 22:08:31 [014] POST /update_jobs/1377112841/increment_metric
2026-05-20T22:08:31.0608297Z   proxy | 2026/05/20 22:08:31 [014] 204 /update_jobs/1377112841/increment_metric
2026-05-20T22:08:31.0685310Z   proxy | 2026/05/20 22:08:31 [016] PATCH /update_jobs/1377112841/mark_as_processed
2026-05-20T22:08:31.1436139Z   proxy | 2026/05/20 22:08:31 [016] 204 /update_jobs/1377112841/mark_as_processed
2026-05-20T22:08:32.3170853Z Failure running container 1e6e460558212e7f4d6667e7aeee2a42c614a662fdb31b7f0b1d87f8ad53402c: Error: Command failed with exit code 1: /bin/sh -c $DEPENDABOT_HOME/dependabot-updater/bin/run update_files
2026-05-20T22:08:32.3873904Z Cleaned up container 1e6e460558212e7f4d6667e7aeee2a42c614a662fdb31b7f0b1d87f8ad53402c
2026-05-20T22:08:32.3957990Z   proxy | 2026/05/20 22:08:32 0/8 calls cached (0%)
2026-05-20T22:08:32.3958810Z   proxy | 2026/05/20 22:08:32 Posting metrics to remote API endpoint
2026-05-20T22:08:32.8457654Z ##[error]Dependabot encountered an error performing the update

Error: The updater encountered one or more errors.

For more information see: https://github.com/Tsingis/bitcoin-web-api/network/updates/1377112841 (write access to the repository is required to view the log)
2026-05-20T22:08:32.8467485Z 🤖 ~ finished: error reported to Dependabot ~
2026-05-20T22:08:32.8645102Z Post job cleanup.
2026-05-20T22:08:32.9962325Z Cleaning up orphan processes
```

--- a/nuget/updater/common.ps1
+++ b/nuget/updater/common.ps1
@@ -57,6 +57,10 @@ function Get-SdkVersionsToInstall([string] $repoRoot, [string[]] $updateDirector
 function Get-DirectoriesMatchingPattern([string] $repoRoot, [string] $pattern) {
     $repoRoot = $repoRoot.Replace("\", "/").TrimEnd("/")
     $pattern = $pattern.Replace("\", "/").Trim("/")
+    if ($pattern -eq ".") {
+        # this handles a common scenario where `$pattern` is initially "/."
+        $pattern = ""
+    }
     $normalizedDirectory = "$repoRoot/$pattern"
     $directoryRegex = "^"
     $includeAnchor = $true

--- a/nuget/updater/test.ps1
+++ b/nuget/updater/test.ps1
@@ -52,6 +52,12 @@ try {
         -expectedSdksToInstall @("1.2.3")
 
     Test-GlobalJsonVersions `
+        -testDirectory "global-json-discovery-root-with-file" `
+        -directories @("/.") `
+        -installedSdks @("8.0.404", "9.0.101") `
+        -expectedSdksToInstall @("1.2.3")
+
+    Test-GlobalJsonVersions `
         -testDirectory "global-json-discovery-none" `
         -directories @("src") `
         -installedSdks @("8.0.404", "9.0.101") `

